lua/pac3/core/client/base_movable.lua

```diff
@@ -184,6 +189,58 @@ function PART:CalcAngles(ang, wpos)
 
 		return self.Angles + (pac.EyePos - wpos):Angle()
 	end
+	
+	if pac.StringFind(self.AimPartName, "NEAREST_LIFE_YAW", true, true) then
+		local nearest_ent = self:GetRootPart():GetOwner()
+		local nearest_dist = math.huge
+		local owner_ent = self:GetRootPart():GetOwner()
+		local part_ent = self:GetOwner()
+		for _,ent in pairs(ents.GetAll()) do
+			if (ent:IsNPC() or ent:IsPlayer()) and ent ~= owner_ent then
+				local dist = (owner_ent:GetPos() - ent:GetPos()):LengthSqr()
+				if dist < nearest_dist then
+					nearest_ent = ent
+					nearest_dist = dist
+				end
+			end
+		end
+		local ang = (nearest_ent:GetPos() - part_ent:GetPos()):Angle()
+		return Angle(0,ang.y,0) + self.Angles
+	end
+	
+	if pac.StringFind(self.AimPartName, "NEAREST_LIFE_POS", true, true) then
+		local nearest_ent = self:GetRootPart():GetOwner()
+		local nearest_dist = math.huge
+		local owner_ent = self:GetRootPart():GetOwner()
+		local part_ent = self:GetOwner()
+		for _,ent in pairs(ents.GetAll()) do
+			if (ent:IsNPC() or ent:IsPlayer()) and ent ~= owner_ent then
+				local dist = (owner_ent:GetPos() - ent:GetPos()):LengthSqr()
+				if dist < nearest_dist then
+					nearest_ent = ent
+					nearest_dist = dist
+				end
+			end
+		end
+		return self.Angles + (nearest_ent:GetPos() - part_ent:GetPos()):Angle()
+	end
+	
+	if pac.StringFind(self.AimPartName, "NEAREST_LIFE", true, true) then
+		local nearest_ent = self:GetRootPart():GetOwner()
+		local nearest_dist = math.huge
+		local owner_ent = self:GetRootPart():GetOwner()
+		local part_ent = self:GetOwner()
+		for _,ent in pairs(ents.GetAll()) do
+			if (ent:IsNPC() or ent:IsPlayer()) and ent ~= owner_ent then
+				local dist = (owner_ent:GetPos() - ent:GetPos()):LengthSqr()
+				if dist < nearest_dist then
+					nearest_ent = ent
+					nearest_dist = dist
+				end
+			end
+		end
+		return self.Angles + ( nearest_ent:GetPos() + Vector(0,0,(nearest_ent:WorldSpaceCenter() - nearest_ent:GetPos()).z * 1.5) - part_ent:GetPos()):Angle()
+	end
 
 	if self.AimPart:IsValid() and self.AimPart.GetWorldPosition then
 		return self.Angles + (self.AimPart:GetWorldPosition() - wpos):Angle()
```

Comment:
Sorry for beintg a bit late and overall absent. I'd rather get to a merge than bikeshedding on how to split something up.

My take on splitting up a PR is based on features, so for example these NEAREST_* "bone" positions could be one feature (don't bother to do this now though). It's a single feature it's independent on the broader combat specific code.